graphium/trainer/losses.py

```diff
@@ -52,22 +52,40 @@ def __init__(
         self.alpha = alpha
         self.softmax = torch.nn.Softmax(dim=1)
 
-    def forward(self, input: Tensor, target: Tensor) -> Tensor:
+    def forward(self, input: Tensor, target: Tensor, nan_targets: Tensor = None) -> Tensor:
         """
         Parameters:
             input: (batch_size x n_classes) tensor of logits predicted for each bracket.
             target: (batch_size) or (batch_size, 1) tensor of target brackets in {0, 1, ..., self.n_brackets}.
         """
 
         target = target.flatten()
+
+        # set input and target with nans to 0s for regression loss
+        input[nan_targets] = 0.0
+        target[nan_targets] = 0.0
```

Comment:
I was worried this wouldn't work on IPU due to dynamic shapes, but it does work!